File: Java/opentraceability/src/main/java/opentraceability/models/masterdata/VocabularyElement.java
package opentraceability.models.masterdata;

import java.util.ArrayList;
import java.util.List;
import opentraceability.interfaces.IMasterDataKDE;
import opentraceability.interfaces.IVocabularyElement;
import opentraceability.interfaces.VocabularyType;
import opentraceability.utility.attributes.OpenTraceabilityExtensionElementsAttribute;

public class VocabularyElement extends IVocabularyElement {
    String id;

    public VocabularyElement()
    {
        epcisType = "urn:epcglobal:epcis:vtype:Unknown";
        vocabularyType = VocabularyType.Unknown;
    }

    @OpenTraceabilityExtensionElementsAttribute
    public List<IMasterDataKDE> kdes = new ArrayList<>();

    @Override
    public String getId() {
        return id;
    }

    @Override
    public void setId(String val) {
        id = val;
    }
}

File: Java/OpenTraceability/src/main/kotlin/opentraceability/models/masterdata/vocabularyelement.kt
package models.masterdata


import interfaces.IMasterDataKDE
import interfaces.IVocabularyElement
import interfaces.VocabularyType
import java.util.*
import models.identifiers.*
import org.json.JSONObject
import utility.EnumUtil
import java.lang.reflect.Type

class VocabularyElement : IVocabularyElement {
    override var ID: String? = null
    override var EPCISType: String? = ""
    override var JsonLDType: String? = null
    override var Context: JSONObject? = null

    var VocabularyType: VocabularyType
        get() {
            var type = VocabularyType.Unknown
            for (t in VocabularyType.values()) {
                if (EnumUtil.getEnumDescription(t).trim().toLowerCase() == EPCISType?.trim().toLowerCase()) {
                    type = t
                }
            }
            return type
        }
        set(value) {
            field = value
        }


    override var KDEs: ArrayList<IMasterDataKDE> = ArrayList<IMasterDataKDE>()

}
